accumulo/src/main/scala/geotrellis/spark/io/accumulo/AccumuloLayerCopier.scala

```diff
@@ -1,55 +1,120 @@
 package geotrellis.spark.io.accumulo
 
-import geotrellis.spark.io.AttributeStore.Fields
-import geotrellis.spark.{Boundable, LayerId}
+import geotrellis.spark._
 import geotrellis.spark.io._
+import geotrellis.spark.io.AttributeStore.Fields
 import geotrellis.spark.io.avro._
-import geotrellis.spark.io.index.KeyIndexMethod
+import geotrellis.spark.io.index._
+import geotrellis.spark.io.json._
 
+import org.apache.avro._
 import org.apache.spark.SparkContext
 import org.apache.spark.rdd.RDD
 import spray.json.JsonFormat
 import scala.reflect.ClassTag
 
-object AccumuloLayerCopier {
-  def apply[K: Boundable: JsonFormat: ClassTag, V: ClassTag, M: JsonFormat](
-   instance   : AccumuloInstance,
-   layerReader: AccumuloLayerReader[K, V, M],
-   layerWriter: AccumuloLayerWriter[K, V, M]
-  )(implicit sc: SparkContext): SparkLayerCopier[AccumuloLayerHeader, K, V, M] = {
-    val writerAttributeStore = layerWriter.attributeStore
-    new SparkLayerCopier[AccumuloLayerHeader, K, V, M](
-      attributeStore = AccumuloAttributeStore(instance.connector),
-      layerReader    = layerReader,
-      layerWriter    = layerWriter
-    ) {
-      def headerUpdate(id: LayerId, header: AccumuloLayerHeader): AccumuloLayerHeader = {
-        val writerHeader = writerAttributeStore.readLayerAttribute[AccumuloLayerHeader](id, Fields.header)
-        header.copy(tileTable = writerHeader.tileTable)
-      }
+class AccumuloLayerCopier(
+  attributeStore: AttributeStore[JsonFormat],
+  layerReader: AccumuloLayerReader,
+  getLayerWriter: LayerId => AccumuloLayerWriter
```

Comment:
What is the thinking behind asking for a reader as function of LayerId? It isn't used internally. Is it user API?